core/src/main/java/brooklyn/entity/basic/AbstractEntity.java

```diff
@@ -962,7 +962,9 @@ public void removeAttribute(AttributeSensor<?> attribute) {
     // -------- CONFIGURATION --------------
 
     @Override
-    public ConfigurationSupportInternal config() {
+    // the concrete type rather than an interface is returned because Groovy subclasses
+    // complain (incorrectly) if we return ConfigurationSupportInternal 
+    public BasicConfigurationSupport config() {
```

Comment:
On balance I'm fine with this, but really would prefer if the impl did not leak as that ties us down for future refactoring. Perhaps we should add an `@Beta` to this method, and add javadoc explaining why (saying that it may change to return a `ConfigurationSupportInternal` in a future release, that being the preferred way to use it).